### ISSUE TITLE
erroneous type sneaks past typer using selectDynamic (requires result type inference)

### ISSUE THREAD

**Issue by scabug** [scala/bug#9558]
```scala
import scala.language.dynamics

abstract class Base { def someInt: Int }
class Test extends Base {
  implicit val foo: String = ""
  class DynWithImplicit extends Dynamic {
    def selectDynamic(k: String)(implicit w: String): String = ???
  }
  
  // error message comes too late (delayed until pickling):
  // expanding the selectDynamic will result in the error being correctly reported during typer  
  def someInt = (new DynWithImplicit).gimmeAnInt 
  //    ^ error: erroneous or inaccessible type 

}
```

**Comment by pabloazul** [comment]
Closing as this appears to be quite obscure and unlikely to be addressed. @adriaanm - please reopen if you think otherwise.

**Comment by joroKr21** [comment]
It's fixed in Scala 2.13
```
         def someInt = (new DynWithImplicit).gimmeAnInt
                        ^
On line 12: error: type mismatch;
        found   : String
        required: Int
```
